app/perfunwind.cpp

```diff
@@ -445,6 +446,7 @@ void PerfUnwind::resolveCallchain()
         symbols->attachDwfl(&m_currentUnwind);
         m_currentUnwind.frames.append(symbols->lookupFrame(ip, isKernel,
                                             &m_currentUnwind.isInterworking));
+        if (!m_currentUnwind.isInterworking) m_currentUnwind.ipStack.append({ip, m_currentUnwind.ipStack.length()});
```

Comment:
this check would need to be applied everywhere, and wrap it in a method such that we can also push the CLI / env var check there

also, style:

```
if (foo)
    bar;
```

Reply:
I added an inline function to do this to ensure it is always checked. Not sure if this is the best practice.